src/Spotflow.InMemory.Azure.ServiceBus/InMemoryServiceBusSessionProcessor.cs

```diff
@@ -0,0 +1,444 @@
+using System.Collections.Concurrent;
+
+using Azure.Messaging.ServiceBus;
+
+using Spotflow.InMemory.Azure.ServiceBus.Resources;
+
+namespace Spotflow.InMemory.Azure.ServiceBus;
+
+public class InMemoryServiceBusSessionProcessor : ServiceBusSessionProcessor
+{
+
+    private readonly int _maxConcurrentCallsPerSession;
+    private readonly int _maxConcurrentSessions;
+    private readonly TimeSpan? _sessionIdleTimeout;
+    private readonly TimeSpan _defaultMaxWaitTime;
+    private readonly SemaphoreSlim _sessionConcurrencySemaphore;
+    private readonly string[] _sessionIds;
+    private readonly InMemoryServiceBusClient _client;
+    private readonly bool _isQueue;
```

Comment:
Consider adding `[MemberNotNullWhen(true, nameof(_queueName))]` and `[MemberNotNullWhen(false, nameof(_topicName), nameof(_subscriptionName))]` attributes to avoid need for the null-forgiving operator `!`.